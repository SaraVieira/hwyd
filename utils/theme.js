import {
  /* colors */
  APPLE_VALLEY,
  BREWED_MUSTARD,
  CREAMY_PEACH,
  OLD_GERANIUM,
  PENCIL_LEAD,
  ROGUE_PINK,
  ROSY_HIGHLIGHT,
  SOFT_BLUE,
  SQUEAKY,
  WHITE,

  /* fonts */
  BOLD_FONT,
  ITALIC_FONT,
  REGULAR_FONT,
  SUBTITLE_FONT,
  TITLE_FONT,
} from './constants';

export default {
  colors: {
    appleValley: APPLE_VALLEY,
    brewedMustard: BREWED_MUSTARD,
    creamyPeach: CREAMY_PEACH,
    oldGeranium: OLD_GERANIUM,
    pencilLead: PENCIL_LEAD,
    roguePink: ROGUE_PINK,
    rosyHighlight: ROSY_HIGHLIGHT,
    softBlue: SOFT_BLUE,
    squeaky: SQUEAKY,
    white: WHITE,
  },
  fonts: {
    bold: BOLD_FONT,
    italic: ITALIC_FONT,
    regular: REGULAR_FONT,
    subtitle: SUBTITLE_FONT,
    title: TITLE_FONT,
  },
};
