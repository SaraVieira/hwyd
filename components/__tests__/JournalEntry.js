import React from 'react';
import { Animated } from 'react-native';
import { ThemeProvider } from 'styled-components';
import { shallow } from 'enzyme';
import 'jest-styled-components';

import JournalEntry from '../JournalEntry';
import theme from '../../utils/theme';

const note = {
  note: 'Today was...',
  result: 'GOOD',
  id: 0,
  date: '03-02-2018',
};

describe('<JournalEntry />', () => {
  it('renders a journal entry', () => {
    const tree = shallow(
      <ThemeProvider theme={theme}>
        <JournalEntry {...note} />
      </ThemeProvider>,
    );
    const root = tree.find(JournalEntry).shallow();
    expect(root).toMatchSnapshot();
  });

  it('renders a journal entry with no title', () => {
    const tree = shallow(
      <ThemeProvider theme={theme}>
        <JournalEntry noTitle {...note} />
      </ThemeProvider>,
    );
    const root = tree.find(JournalEntry).shallow();
    expect(root).toMatchSnapshot();
  });

  it('renders an animated journal entry', () => {
    const tree = shallow(
      <ThemeProvider theme={theme}>
        <JournalEntry
          animatedValue={new Animated.Value(0)}
          itemIndex={0}
          {...note}
        />
      </ThemeProvider>,
    );
    const root = tree.find(JournalEntry).shallow();
    expect(root).toMatchSnapshot();
  });
});
