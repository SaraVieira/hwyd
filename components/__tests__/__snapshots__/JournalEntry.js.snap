// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JournalEntry /> renders a journal entry 1`] = `
<Styled(AnimatedComponent)
  style={false}
>
  <Styled(View)>
    <Styled(Text)>
      March 2nd, 2018
    </Styled(Text)>
    <Styled(Icon)
      color="#cf6a87"
      name="emoji-happy"
      size={16}
    />
    <Styled(View) />
    <Styled(AnimatedComponent)
      style={false}
    >
      Today was...
    </Styled(AnimatedComponent)>
  </Styled(View)>
</Styled(AnimatedComponent)>
`;

exports[`<JournalEntry /> renders a journal entry with no title 1`] = `
<Styled(AnimatedComponent)
  style={false}
>
  <Styled(View)>
    <Styled(AnimatedComponent)
      style={false}
    >
      Today was...
    </Styled(AnimatedComponent)>
  </Styled(View)>
</Styled(AnimatedComponent)>
`;

exports[`<JournalEntry /> renders an animated journal entry 1`] = `
<Styled(AnimatedComponent)
  style={
    Object {
      "transform": Array [
        Object {
          "scale": 1.1,
        },
      ],
    }
  }
>
  <Styled(View)>
    <Styled(Text)>
      March 2nd, 2018
    </Styled(Text)>
    <Styled(Icon)
      color="#cf6a87"
      name="emoji-happy"
      size={16}
    />
    <Styled(View) />
    <Styled(AnimatedComponent)
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      Today was...
    </Styled(AnimatedComponent)>
  </Styled(View)>
</Styled(AnimatedComponent)>
`;
